994.py

```diff
@@ -0,0 +1,40 @@
+class Solution:
+    def orangesRotting(self, grid: List[List[int]]) -> int:
+        R, C = len(grid), len(grid[0])
+        queue = deque([])
+        fresh_oranges = 0
+
+        # Find all the rotten oranges.
+        for i in range(R):
+            for j in range(C):
+                if grid[i][j] == 2:
+                    queue.append((i, j))
+                elif grid[i][j] == 1:
+                    fresh_oranges += 1
+
+        if fresh_oranges == 0:
+            return 0
+
+        minute = 0
+        DIRS = ((1, 0), (-1, 0), (0, 1), (0, -1))
+
+        while queue:
+            queue_length = len(queue)
+
+            for _ in range(queue_length):
```

Comment:
I can see this is the template you used when solving the tree-based BFS problems, and you are able to apply it here.

Reply:
Sure, I plan to do that soon.